twyla/kubedeploy/kube.py

```diff
@@ -88,11 +90,55 @@ def type_name_from_data(self, data):
         return ''.join(parts)
 
 
+    def api_name_from_object(self, obj):
+        # apps/v1beta1
+        api_version = obj.api_version
+        # ['Apps', 'V1beta1']
+        parts = [part.capitalize() for part in api_version.split('/')]
+        # ['Apps', 'V1beta1']
+        parts.append('Api')
+        # 'AppsV1beta1Api'
+        return ''.join(parts)
+
+
+    def api_from_object(self, obj):
+        api_name = self.api_name_from_object(obj)
+        try:
+            api = getattr(kubernetes.client, api_name)()
+        except AttributeError:
+            api = getattr(kubernetes.client, 'Core{}'.format(api_name))()
+
+        return api
+
+
     def parse_data(self, data: dict):
         kubernetes_type = self.type_name_from_data(data)
         api_client = kubernetes.client.ApiClient()
         res = Res(data=json.dumps(data))
-        return api_client.deserialize(res, kubernetes_type)
+        obj = api_client.deserialize(res, kubernetes_type)
+
+        # Hackaround missing intOrString support in the python client
+        # TODO: check how to fix that upstream
+        if kubernetes_type.endswith('Deployment'):
+            try:
+                int_val = int(obj.spec.strategy.rolling_update.max_surge)
+                obj.spec.strategy.rolling_update.max_surge = int_val
+            except:
```

Comment:
Catching every exception here does not look like a good idea. If you want to simply ignore the case where the `max_surge` attribute cannot be turned into an int, the exception you should catch is `ValueError`.